### PR TITLE
PGF - Elated Bolide Class Assault Lander

### PR DESCRIPTION
**Pull request by Erikafox**
## About The Pull Request
Adds the Elated Bolide Class and assorted fluff that I made for it. 

StrongDMM-2024-09-28 22 47 07
StrongDMM-2024-09-28 22 47 08

The Elated Bolide is named for the excitement that it's crew experiences when hitting a hostile atmosphere, and the way that a particularly hard entry can appear from the ground. 
In the frontier, the Bolide-Class is used both in exploration and civilization building (imperialism) with the marines being equipped with a surplus of equipment to quickly break down and salvage unwanted structures (enemy fortifications) while building colonial infrastructure, such as concrete roads (bunker complexes). 
6 Marines are carried, 4 of which are marine engineers. The traditional assignment is splitting the marines into 2 squads of 3, although in the end it is up to the Captain and Marine LT to decide how the teams are organized. Several fluff papers have been included to try and encourage roleplay aboard the vessel, as well as a degree of paperpushing. 

The Bolide is well defended from the sides thanks to it's Etherbor Point Defense Systems, with the rear mounts carrying a lighter deterrent turret. 
The Bolide is also well armed thanks to it's armory, able to stock all 6 marines (assuming that some marines are willing to carry Tools of some flavor instead of a large gun), and contains some specialist gear for sensitive explosives defusal. And explosives planting.

TLDR:
- High powered turrets near the cargo bay
- Surplus of basic engineering equipment
- Decently sized armory
- Decently sized cargo bay
- Lack in medical, advanced engineering equipment.
- Lots of demolition tools. 
<details><summary>Pictures</summary>
<p>

dreamseeker_sJpVpK4vye
dreamseeker_pAFuv4kWM6
dreamseeker_KjBGWSFeXV
dreamseeker_hVrb3EOixZ

</p>
</details> 

## Why It's Good For The Game
's PGF content. Needs a bit more decor though but I'm at a loss for how right now. Technically completely playable.

## Changelog

:cl:
add: A new PGFN vessel has entered the frontier - the Elated Bolide class makes a bright entry into the local system.
/:cl:

### CODE REVIEW

**Comment by Erikafox** [comment]
it's not going to compile Properly before #3386 is merged because i gave the turrets an id var. But it does load. It does work. It just runtimes.

**Review by meemofcourse — CHANGES_REQUESTED** [review]
imagen
THE NAVY MUST NEVER MERGE WITH THE MARINES!!! (separate navy and marine wear please)

imagen
there's already a crayon ration on the ship so im guessing its intentional

imagen
this is clever but i'm not sure if i agree entirely with it. up to you (or another maptainer i guess)

imagen
there is ! ! UNSECURED URANIUM ! !

imagen
move the ration in the engi locker

imagen
yea

imagen
i feel as though the captain should get headwear

imagen
the SMESs don't seem to be connected

imagen
i disagree on having Military Grade Explosives in the explosive storage instead of in the armory

**Comment by Erikafox** [comment]
>explosives storage

It's locked to the armory code regardless. I figured that'd be a better spot to put them as an 'engineering' tool, versus the armory which is weaponry.

>handrails

Those are supposed to be for a quick marine disembark and the crates being there is coincidence so the room isn't cratehell.

Rest should be handled

**Review by meemofcourse — APPROVED** [review]
sure why not lets tm it

**Comment by fighterslam** [comment]
Feedback (I AM NOT A MAINTAINER OR MAPTAINER, SO DO NOT TAKE THIS AS THE WORD OF GOD <3)

-- !!__IMPORTANT__!! Set docking port "preferred_direction" to "4" so it flies straight and does not reverse itself during transit. 

- Some of the airlocks lack labelling which, while not a huge deal, does make the ship feel less baked. The same for the Shutters, it's good to be able to look at a button & corresponding shutters to know that they're linked.

- The ship lacks a modcomp/ ID console, which is useful for things like chat client, notekeeper, and importantly, the ID console.
I believe there's a program you can download off modcomps to check ship alerts too, so you can probably change the emergency alert console in the bridge to a ModComp. 

- The external air cycler forces you to go through Medbay in order to access it. While this does work, it's a little bit awkward. The only potential solution I could think of is moving the Showers & Captain's Quarters up a bit, while moving the airlock between the Bridge & CPT's quarters. This would be a significant rearrangement though and I don't know how it'd feel in comparison to its current state. Just something to look into.

- Crate Shelves: The crate shelves in the cargo bay are smaller than normal, only holding 2 crates instead of the traditional 3. I recommend making the 2-tall variant craftable with metal sheets, that way they can be replaced or moved around if the crew wants to. Cool addition though! Thanks for adding.

**Comment by DIB-DOG** [comment]
Bringing in some more feedback, also not a maptainer so take it as you will.

1. The engine rooms as far as I saw don't have that standard paper that comes with most ships with combustion engines, adding that sheet is a massive aid to new players or engineers who aren't familiar with the concepts of combustion engines or how to make a good gas mix.

engine paper


2. The marine armory only contains 4 survival knives. As a branch that prides itself on its close combat proficiency, it seems wrong that each marine does not have their own blade. This could of course be for balance or some other reason, but from a purely IC perspective, I think each marine should have a knife.

armory


3. I have a couple suggestions in regard to the uniform and armor lockers at the bottom of the Armory.

- I believe the uniform locker only contains 3 pairs of gloves, for a team of 6 total marines. The uniform locker also lacks any of the PGFMC flap caps which I must say, look quite nice on the marines.

- The amor lockers contains 6 coated Raksha Vests and 6 Raksha Helmets. On the Crying sun the Coated vest is reserved as an option for the Marine Lieutenant, but for the context of this vessel, I do really like the idea of more of the marines wearing the coated vest. However, I wanted to suggest perhaps placing 4 coated vests and 2 standard vests in the locker instead. The Marine Lieutenant on this ship has a standard vest as well, so allowing for the 2 normal marines to wear standard vests may serve as another interesting distinction between the Marine Pioneers and the Marine Riflemen. Just a thought

Marine clothing


4. The Marine Lieutenant's locker is missing a few things and has another which it doesn't need.

- The locker lacks a Marine Officer's flap cap as well as a helmet for the Lieutenant. If anyone should be demonstrating the importance of wearing protective equipment, it should be the Lieutenant.

- The locker has a blue silkenweave jacket which is a navy uniform item. While it's not vital that it be removed, it does break the standard of not mixing Marine and Navy apparel.

Marine Sergeant room


5. In the bridge, the buttons for controlling the ship's shutters are in confusing places and may be better changing spots. The button for closing the bridge window shutters is tucked behind a wall while the button to open the medbay shutter is sitting right in front of the Bridge window shutters. Nothing terrible, but when on the bridge, it feels like the two buttons should trade places as to be more natural.

Bridge


6. I think it's only a TM rn but adding some bunk-beds to this room if that PR gets greenlit would be very cool. Also, I don't think I saw a radio for music anywhere on board so adding one in here would be pretty nice if you think it fits the vibe. The crying sun has one so why not, right?

Screenshot 2024-09-29 004237


7. The whistles are normal sized. These things are huge when I put them in my bag, but I can fit it in my mouth no problem? Haha, in all seriousness though, not sure if this would be within scope for the project but these things really ought to be small if not tiny sized.

Screenshot 2024-09-29 003527
Screenshot 2024-09-29 003533


With all that said, that's all I have to report! I love the ship and really look forward to using it in the future. I really dig the design and scale of it. So many cool little rooms and details loaded onto here, really awesome work.

**Comment by Erikafox** [comment]
Thanks for the feedback. It should all be worked in.
The big screencaps have been updated with the Current Design.

**Comment by DIB-DOG** [comment]
Just a bit more feedback after having played the Bolide on Wednesday.

bridge idea

1. When playing as the captain and letting my helmsman pilot the vessel, I found that I was often asked to take a look at details on the ship's console or simply wanted to take a peek at our progress while midflight. To do this however, the helmsman had to either unbuckle from their seat and step to the side, or I would have to crawl to pass them. I believe this could be remedied by placing two seats at the control console, one on each end however I don’t think it’s the best solution.

Viewscreen idea
console 2

Alternatively, adding a ship viewscreen to the bridge would alleviate these issues and maintain the very cool cockpit vibe that the helm console has. Just as a proof of concept I placed one where there was some free space and it doesn't look too bad I feel. Of course, this is just an idea but I do feel like adding a ship viewscreen some places around the ship like the PGFN Crying Sun has would be nice.


screwdrivers

2. Just a little equipment nitpick on the marine end. Not a necessity, but convenient and logical I feel. I suppose most if not all of the marine pioneers will probably have a screwdriver included in their kit anyhow, so if you think it no concern I’ll leave it to you. Just some food for thought.

File: code/game/objects/items/storage/boxes.dm [hunk 1]
```diff
@@ -768,6 +776,17 @@
...
+	name = "box of smoke grenades"
+	desc = "Used for rapidly laying cover."
+	illustration = "grenade"
+
+/obj/item/storage/box/smokebombs/PopulateContents()
+	for(var/i in 1 to 7)
+		new /obj/item/grenade/smokebomb(src)
+
+
+
```
Comment (FalloutFalcon) [inline code/game/objects/items/storage/boxes.dm]: 3 tabs... so rich..

**Review by FalloutFalcon — APPROVED** [review]

**Comment by Apogee-dev** [comment]
The decaling style could stand to cut a little closer to the example set by the Crying Sun i think, particularly the hallway decals. The green fancy spline + steel grid combo is something only the crying sun does at the moment, while the border decal set you used here with standard dark tiles is probably the most widely used set of decals in the game right now. 
There's also the fact that it's just much less colorful in general, but we already had that conversation about the other PGF ship you're working on. You _could_ switch the colors in a couple rooms, such as having bottle green rather than lime green in the lieutenant's office and using some standard grey tiles in engineering (probably just on the upper level of the main room).
Anyway, if possible, I'd like you to copy the Crying Sun's decaling style for the hallways at a minimum, because those looked fantastic on the Sun and they'll look fantastic here.
